report_docx/report/report_docx.py

```diff
@@ -0,0 +1,306 @@
+        self.uid = uid
+        report_obj = self.pool.get('ir.actions.report.xml')
+        report_ids = report_obj.search(
+            cr, uid, [('report_name', '=', self.name[7:])], context=context)
+
+        if report_ids:
+            report_xml = report_obj.browse(
+                cr, uid, report_ids[0], context=context)
+            self.title = report_xml.name
+            if report_xml.report_type == 'docx':
+                return self.create_source_docx(cr, uid, ids, context)
+        return super(ReportDocx, self).create(cr, uid, ids, data, context)
+
+    def create_source_docx(self, cr, uid, ids, context=None):
+        data = self.generate_docx_data(cr, uid, ids, context)
+
+        tmp_folder_name = '/tmp/docx_to_pdf/'
+        output_type = self._get_output_type(cr, uid, context)
+        output_report = {
+            'pdf': 'report.pdf',
+            'docx': 'report.docx'
+        }
+
+        self._delete_temp_folder(tmp_folder_name)
+        self._create_temp_folder(tmp_folder_name)
+
+        self._generate_reports(
+            cr, uid, context, tmp_folder_name, data,
+            output_type, output_report)
+
+        report = self._get_convert_file(
+            tmp_folder_name, output_report[output_type])
+
+        self._delete_temp_folder(tmp_folder_name)
+
+        return (report, output_type)
+
+    def generate_docx_data(self, cr, uid, ids, context):
+        """
+            Override this method to pass your own data to the engine.
+            The return value of this module should be a list with
+            report data.
+        """
+        return [{}]
+
+    def _generate_reports(
+        self, cr, uid, context, tmp_folder_name,
+        datas, output_type, output_report
+    ):
+        if "pdf" == output_type:
+            self._generate_pdf_reports(
+                cr, uid, context, tmp_folder_name, datas,
+                output_type, output_report)
+            return
+
+        self._generate_doc_reports(
+            cr, uid, context, tmp_folder_name, datas,
+            output_type, output_report)
+
+    def _generate_pdf_reports(
+        self, cr, uid, context, tmp_folder_name,
+        datas, output_type, output_report
+    ):
+        count = 0
+        for data in datas:
+            self._convert_single_report(
+                cr, uid, context, tmp_folder_name, count, data, output_type)
+            count = count + 1
+
+        self._combine_pdf_files(
+            tmp_folder_name, output_report[output_type])
+
+    def _generate_doc_reports(
+        self, cr, uid, context, tmp_folder_name,
+        datas, output_type, output_report
+    ):
+        if len(datas) > 1:
+            raise Exception(_(
+                "Multiple docx files export doesn't support. " +
+                "Please export single docx file."))
+        temp_report = self._convert_single_report(
+            cr, uid, context, tmp_folder_name, 0, datas[0], output_type)
+
+        cmd = "mv %s %s" % (
```

Comment:
IMO this should be possible using the Python OS interface directly, without the need for a shell.